c++20 compiling issue

```
In file included from src/dEploidIO.hpp:37,
                 from src/dEploid.cpp:27:
src/param.hpp:82:20: error: expected unqualified-id before ‘)’ token
   82 |     Parameter <T> () { }
      |                    ^
src/param.hpp:83:21: error: expected ‘)’ before ‘value’
   83 |     Parameter <T> (T value) {
      |                   ~ ^~~~~~
      |                     )
src/param.hpp:87:20: error: expected unqualified-id before ‘const’
   87 |     Parameter <T> (const Parameter <T> &currentParam) {
      |                    ^~~~~
src/param.hpp:87:20: error: expected ‘)’ before ‘const’
   87 |     Parameter <T> (const Parameter <T> &currentParam) {
      |                   ~^~~~~
      |                    )
make: *** [Makefile:1795: src/dEploid-dEploid.o] Error 1
```

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -42,7 +42,7 @@ jobs:
       uses: actions/setup-node@v3
       with:
         node-version: 16.x
-            
+
     - uses: actions/setup-python@v3
       with:
         python-version: 3.8
@@ -123,10 +123,10 @@ jobs:
     - name: docs
       run:  if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile;  fi
 
-    - name: run coverage
-      if: matrix.name == 'ubuntu-gcc-10'
-      run: |
-         tests/test-coverage-coverall.sh
+#    - name: run coverage
+#      if: matrix.name == 'ubuntu-gcc-10'
+#      run: |
+#         tests/test-coverage-coverall.sh
 
 #  See: https://github.com/marketplace/actions/coveralls-github-action
     - name: Coveralls

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -123,10 +123,10 @@ jobs:
     - name: docs
       run:  if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile;  fi
 
-#    - name: run coverage
-#      if: matrix.name == 'ubuntu-gcc-10'
-#      run: |
-#         tests/test-coverage-coverall.sh
+    - name: run coverage
+      if: matrix.name == 'ubuntu-gcc-10'
+      run: |
+         tests/test-coverage-coverall.sh
 
 #  See: https://github.com/marketplace/actions/coveralls-github-action
     - name: Coveralls

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -88,7 +88,7 @@ jobs:
         echo "::set-output name=timestamp::${stamp}"
 
     - name: ccache cache files
-      uses: actions/cache@v2
+      uses: actions/cache@v4
       with:
          path: ~/.ccache
          key: ${{ matrix.name }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -94,7 +94,12 @@ jobs:
          key: ${{ matrix.name }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
          restore-keys: |
            ${{ matrix.name }}-ccache-
-
+    
+    - name: Use Node.js 16.x
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16.x
+        
     - name: configure
       run: ./bootstrap
 
@@ -134,4 +139,6 @@ jobs:
       uses: coverallsapp/github-action@v2.3.4
       with:
         github-token: ${{ secrets.GITHUB_TOKEN }}
-      run: coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'
+      # run: coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'
+
+      

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -123,10 +123,10 @@ jobs:
     - name: docs
       run:  if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile;  fi
 
-    - name: run coverage
-      if: matrix.name == 'ubuntu-gcc-10'
-      run: |
-         tests/test-coverage-coverall.sh
+    # - name: run coverage
+    #  if: matrix.name == 'ubuntu-gcc-10'
+    #  run: |
+    #     tests/test-coverage-coverall.sh
 
 #  See: https://github.com/marketplace/actions/coveralls-github-action
     - name: Coveralls
@@ -134,4 +134,4 @@ jobs:
       uses: coverallsapp/github-action@v2.3.4
       with:
         github-token: ${{ secrets.GITHUB_TOKEN }}
-      #run: coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'
+      run: coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -94,12 +94,12 @@ jobs:
          key: ${{ matrix.name }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
          restore-keys: |
            ${{ matrix.name }}-ccache-
-    
+
     - name: Use Node.js 16.x
       uses: actions/setup-node@v3
       with:
         node-version: 16.x
-        
+
     - name: configure
       run: ./bootstrap
 
@@ -134,11 +134,11 @@ jobs:
     #     tests/test-coverage-coverall.sh
 
 #  See: https://github.com/marketplace/actions/coveralls-github-action
-    - name: Coveralls
-      if: matrix.name == 'ubuntu-gcc-10'
-      uses: coverallsapp/github-action@v2.3.4
-      with:
-        github-token: ${{ secrets.GITHUB_TOKEN }}
+    # - name: Coveralls
+    #  if: matrix.name == 'ubuntu-gcc-10'
+    #  uses: coverallsapp/github-action@v2.3.4
+    #  with:
+    #    github-token: ${{ secrets.GITHUB_TOKEN }}
       # run: coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'
 
-      
+

--- a/Makefile.am
+++ b/Makefile.am
@@ -13,7 +13,7 @@ bin_PROGRAMS = dEploid dEploid_dbg
 man1_MANS = docs/_build/man/dEploid.1
 
 TESTS = unit_tests io_unit_tests
-check_PROGRAMS = unit_tests_1 unit_tests_2 unit_tests dEploid_dbg io_unit_tests current_unit_tests # dEploid_prof
+check_PROGRAMS = unit_tests dEploid_dbg io_unit_tests current_unit_tests # dEploid_prof
 PROG = DEPLOID
 
 common_flags = -std=c++2a -Isrc/ -Isrc/codeCogs/ -Isrc/vcf/src/ -DDEPLOIDVERSION=\"${DEPLOIDVERSION}\" -DLASSOVERSION=\"${LASSOVERSION}\" -DCOMPILEDATE=\"${COMPILEDATE}\" -Wall  -Wextra
@@ -57,36 +57,18 @@ dEploid_LDADD = $(common_LDADD)
 dEploid_dbg_LDADD = $(common_LDADD)
 dEploid_prof_LDADD = $(common_LDADD)
 
-unit_tests_1_SOURCES_0 = tests/unittest/test_ibd.cpp \
-					 tests/unittest/test_panel.cpp \
-					 tests/unittest/test_utilities.cpp
-
-unit_tests_2_SOURCES_0 = tests/unittest/test_mcmc.cpp \
+unit_tests_SOURCES = $(common_src) \
+					 tests/unittest/test_runner.cpp \
+					 tests/unittest/test_ibd.cpp \
 					 tests/unittest/test_updateSingleHap.cpp \
+					 tests/unittest/test_utilities.cpp \
+					 tests/unittest/test_panel.cpp \
+					 tests/unittest/test_mcmc.cpp \
 					 tests/unittest/test_updatePairHap.cpp \
 					 tests/unittest/test_workflow.cpp
 
-unit_tests_1_SOURCES = $(common_src) \
-					 $(unit_tests_1_SOURCES_0) \
-					 tests/unittest/test_runner.cpp
-
-unit_tests_2_SOURCES = $(common_src) \
-					 $(unit_tests_2_SOURCES_0) \
-					 tests/unittest/test_runner.cpp
-
-unit_tests_SOURCES = $(common_src) \
-					 $(unit_tests_1_SOURCES_0) \
-                     $(unit_tests_2_SOURCES_0) \
-					 tests/unittest/test_runner.cpp
-
 unit_tests_CXXFLAGS = $(common_flags) -DNDEBUG -DUNITTEST -Wno-write-strings --coverage
 unit_tests_LDADD    = -lcppunit -ldl $(common_LDADD)
-
-unit_tests_1_CXXFLAGS = $(common_flags) -DNDEBUG -DUNITTEST -Wno-write-strings --coverage
-unit_tests_1_LDADD    = -lcppunit -ldl $(common_LDADD)
-
-unit_tests_2_CXXFLAGS = $(common_flags) -DNDEBUG -DUNITTEST -Wno-write-strings --coverage
-unit_tests_2_LDADD    = -lcppunit -ldl $(common_LDADD)
 
 io_unit_tests_SOURCES = $(common_src) \
 					 tests/unittest/test_runner.cpp \
@@ -116,4 +98,3 @@ if ENABLE_BASH_COMPLETION
 bashcompletiondir = $(BASH_COMPLETION_DIR)
 dist_bashcompletion_DATA = src/dEploid_completion.sh
 endif
-

--- a/Makefile.am
+++ b/Makefile.am
@@ -16,7 +16,7 @@ TESTS = unit_tests io_unit_tests
 check_PROGRAMS = unit_tests dEploid_dbg io_unit_tests current_unit_tests # dEploid_prof
 PROG = DEPLOID
 
-common_flags = -std=c++17 -Isrc/ -Isrc/codeCogs/ -Isrc/vcf/src/ -DDEPLOIDVERSION=\"${DEPLOIDVERSION}\" -DLASSOVERSION=\"${LASSOVERSION}\" -DCOMPILEDATE=\"${COMPILEDATE}\" -Wall  -Wextra
+common_flags = -std=c++20 -Isrc/ -Isrc/codeCogs/ -Isrc/vcf/src/ -DDEPLOIDVERSION=\"${DEPLOIDVERSION}\" -DLASSOVERSION=\"${LASSOVERSION}\" -DCOMPILEDATE=\"${COMPILEDATE}\" -Wall  -Wextra
 
 common_LDADD = -lz
 

--- a/src/param.hpp
+++ b/src/param.hpp
@@ -79,12 +79,12 @@ friend class DEploidIO;
     }
 
  public:
-    Parameter <T> () { }
-    Parameter <T> (T value) {
+    Parameter() { }
+    Parameter(T value) {
         this->init(value);
     }
     T getValue() const;
-    Parameter <T> (const Parameter <T> &currentParam) {
+    Parameter(const Parameter<T> &currentParam) {
         this->makeCopy(currentParam);
     }
     ~Parameter() { }

--- a/tests/test-coverage-coverall.sh
+++ b/tests/test-coverage-coverall.sh
@@ -4,8 +4,8 @@
 make -mj
 
 # Generate html report
-lcov --ignore-errors empty --base-directory . --directory . --zerocounters -q
+lcov --ignore-errors version --base-directory . --directory . --zerocounters -q
 make check -mj
-lcov --ignore-errors empty --base-directory . --directory . -c -o coverage/lcov.info
+lcov --ignore-errors version --base-directory . --directory . -c -o coverage/lcov.info
 # --rc lcov_branch_coverage=1 option will turn on branch check
-lcov --ignore-errors empty --remove coverage/lcov.info "/usr*" "src/codeCogs/*" "src/vcf/*" "src/lasso/*" "src/export/*" "src/gzstream/*" "tests/unittest/*" -o coverage/lcov.info # remove output for external libraries
+lcov --ignore-errors version --remove coverage/lcov.info "/usr*" "src/codeCogs/*" "src/vcf/*" "src/lasso/*" "src/export/*" "src/gzstream/*" "tests/unittest/*" -o coverage/lcov.info # remove output for external libraries

